examples/vision/super_resolution_sub_pixel.py

```diff
@@ -0,0 +1,513 @@
+
+input_transform = transform_input(y, crop_size, upscale_factor)
+print("Input image size after transformation is ", input_transform.size)
+display(input_transform)
+
+target_transform = transform_target(y, crop_size)
+print("Target image size after transformation is ", target_transform.size)
+display(target_transform)
+
+"""
+## Use `Sequence` class to load and vectorize batches of data.
+"""
+
+
+class BSDS500(tf.keras.utils.Sequence):
+    def __init__(
+        self,
+        input_img_size,
+        target_img_size,
+        crop_size,
+        batch_size,
+        input_img_paths,
+        upscale_factor=3,
+        transform_target=None,
+        transform_input=None,
+        channels=1,
+    ):
+        super(BSDS500).__init__()
+        self.batch_size = batch_size
+        self.input_img_size = input_img_size
+        self.target_img_size = target_img_size
+        self.crop_size = crop_size
+        self.upscale_factor = upscale_factor
+        self.input_img_paths = input_img_paths
+        self.transform_input = transform_input
+        self.transform_target = transform_target
+        self.channels = channels
+
+    def __len__(self):
+        return len(self.input_img_paths) // self.batch_size
+
+    def __getitem__(self, idx):
+        """Return tuple (input, target) with corresponding #idx"""
+        i = idx * self.batch_size
+        batch_input_img_paths = self.input_img_paths[i : i + self.batch_size]
+        batch_target_img_paths = self.input_img_paths[i : i + self.batch_size]
+        # Build the dataset as `batch_size, img_size[0], img_size[1], channels`
+        x = np.zeros(
+            (self.batch_size,) + self.input_img_size + (self.channels,), dtype="float32"
+        )
+        for j, path in enumerate(batch_input_img_paths):
+            img, _, _ = load_img(path).convert("YCbCr").split()
+            img = self.transform_input(img, self.crop_size, self.upscale_factor)
+            img = tf.keras.preprocessing.image.img_to_array(img)
+            # Scale to [0, 1]
+            img /= 255.0
+            x[j] = img
+
+        y = np.zeros(
+            (self.batch_size,) + self.target_img_size + (self.channels,),
+            dtype="float32",
+        )
+        for j, path in enumerate(batch_target_img_paths):
+            img, _, _ = load_img(path).convert("YCbCr").split()
+            img = self.transform_target(img, self.crop_size)
+            img = tf.keras.preprocessing.image.img_to_array(img)
+            # Scale to [0, 1]
+            img /= 255.0
+            y[j] = img
+
+        return x, y
+
+
+"""
+## Define the model
+"""
+
+crop_size = calculate_crop_size(crop_size, upscale_factor)
+
+input_img_size = (crop_size // upscale_factor, crop_size // upscale_factor)
+target_img_size = (crop_size, crop_size)
+
+channels = 1
+
+batch_size = 32
+
+# input_shape = (input_img_size[0], input_img_size[1], channels)
+
+from tensorflow.keras.layers import Conv2D
+
+
+class ESPCN(tf.keras.Model):
```

Comment:
Please use the functional API to define this model, it will be less verbose, more readable, and less error-prone.
```python
conv_args = {"activation": "relu", "kernel_initializer": "orthogonal", "padding": "same"}
inputs = keras.Inputs(shape=input_img_size + (1,))
x = layers.Conv2D(64, 5, **conv_args)(inputs)
x = layers.Conv2D(64, 3, **conv_args)(x)
x = layers.Conv2D(64, 3, **conv_args)(x)
x = layers.Conv2D(channels * upscale_factor ** 2, 3, **conv_args)(x)
outputs = tf.nn.depth_to_space(x, upscale_factor)
model = keras.Model(inputs, outputs)
model.summary()
```

Did you really mean `channels * upscale_factor ** 2`, or `(channels * upscale_factor) ** 2`?

Reply:
It should be `channels * (upscale_factor ** 2)`.